components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+	public boolean addPolicy(Policy policy) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.addPolicy(policy);
+			response.setStatus(Response.Status.CREATED.getStatusCode());
+			if (log.isDebugEnabled()) {
+				log.debug("Policy has been added successfully.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@POST
+	@Path("/active-policy")
+	@Produces("application/json")
+	public boolean addActivePolicy(Policy policy) {
+
+		policy.setActive(true);
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.addPolicy(policy);
+			response.setStatus(Response.Status.CREATED.getStatusCode());
+			if (log.isDebugEnabled()) {
+				log.debug("Policy has been added successfully.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Produces("application/json")
+	public Policy[] getAllPolicies() {
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint policyAdministratorPoint = policyManagerService.getPAP();
+			List<Policy> policies = policyAdministratorPoint.getPolicies();
+			return policyAdministratorPoint.getPolicies().toArray(new Policy[policies.size()]);
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return null;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return null;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Produces("application/json")
+	@Path("/{id}")
+	public Policy getPolicy(@PathParam("id") int policyId) {
```

Comment:
why do not we simply return the javax.ws.rs.core.Response object with policy payload built into its body here and make the return type consistent across all the APIs?